tests/config/test_config_pyproject_toml.py

```diff
@@ -0,0 +1,70 @@
+from pathlib import Path
+from typing import List
+from unittest.mock import patch
+
+import pytest
+
+from pylint import run_pylint
+
+HERE = Path(__file__).parent
+
+
+@pytest.mark.parametrize(
+    "pyproject_toml_path,expected_results",
+    [
+        [
+            "issue_4580",
+            {
+                "basic_example.toml": {"code": 0},
+                "correct_basic_name_group.toml": {"code": 0},
+                "correct_import_preferred_module.toml": {"code": 0},
+                "empty_list.toml": {
+                    "code": 2,
+                    "out": "encountered 'load-plugins' : '[]'",
+                },
+                "invalid_data_for_basic.toml": {
+                    "code": 2,
+                    "out": "encountered 'load-plugins' : '[]'",
+                },
+                "invalid_data_for_import.toml": {"code": 0},
+                "rich_types.toml": {"code": 0},
+                "top_level_disable.toml": {
+                    "code": 2,
+                    "out": "encountered 'disable' : 'logging-not-lazy,",
+                },
+            },
+        ],
+        [
+            "issue_4746",
+            {
+                "loaded_plugin_does_not_exists.toml": {
+                    "code": 2,
+                    "out": "'pylint_websockets' is impossible to load",
+                }
+            },
+        ],
+    ],
+)
+def test_config_pyproject_toml(pyproject_toml_path, expected_results, capsys):
+    pyproject_toml_paths: List[Path] = (HERE / pyproject_toml_path).iterdir()
+    for toml_path in pyproject_toml_paths:
+        expected_result = expected_results[toml_path.name]
+        file_to_lint = str(HERE / "file_to_lint.py")
+        with patch(
+            "sys.argv",
+            ["pylint", "--rcfile", str(toml_path), file_to_lint, "-r", "n"],
+        ):
+            try:
+                run_pylint()
+            except SystemExit as ex:
+                out, err = capsys.readouterr()
+                msg = f"Wrong result with configuration {toml_path}"
+                if "out" not in expected_result:
+                    assert not out, msg
+                else:
+                    assert expected_result["out"] in out, msg
```

Comment:
Also, I think we should check for the full message.
Checking for `"encountered 'disable' : 'logging-not-lazy,",` does not really tell me what the expected output of the message is.

Reply:
Resolving in favor of this discussion : https://github.com/PyCQA/pylint/pull/4720/files#r744555630